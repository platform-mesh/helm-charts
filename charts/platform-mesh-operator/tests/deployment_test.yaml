suite: operator
templates:
  - deployment.yaml
  - service-account.yaml
  - cluster-role.yaml
  - cluster-rolebinding.yaml
  - role.yaml
  - rolebinding.yaml
values:
  - ../test-values.yaml
chart:
  version: 1.0.0
  appVersion: 1.0.0
release:
  name: platform-mesh-operator
tests:
- it: operator match the snapshot
  set:
    deployment:
      replicas: 0
  asserts:
  - matchSnapshot: {}
- it: deployment with tracing
  template: deployment.yaml
  set:
    tracing:
      enabled: true
      collector:
        endpoint: test:4317
  asserts:
    - matchSnapshot: {}
- it: has all health checks
  template: deployment.yaml
  asserts:
    - equal:
        path: spec.template.spec.containers[0].livenessProbe.httpGet.path
        value: /healthz
    - equal:
        path: spec.template.spec.containers[0].readinessProbe.httpGet.path
        value: /readyz
    - equal:
        path: spec.template.spec.containers[0].startupProbe.httpGet.path
        value: /readyz

- it: multi-pod leader election
  template: deployment.yaml
  set:
    deployment:
      replicas: 3
    operator:
      leaderElect: true
  asserts:
    - equal:
        path: spec.replicas
        value: 3
    - equal:
        path: spec.template.spec.containers[0].args[1]
        value: "--leader-elect"
