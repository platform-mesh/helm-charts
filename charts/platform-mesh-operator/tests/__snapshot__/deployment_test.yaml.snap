deployment with tracing:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: platform-mesh-operator
      namespace: NAMESPACE
    spec:
      replicas: 1
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: platform-mesh-operator
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          annotations:
            sidecar.istio.io/inject: "true"
          labels:
            app: platform-mesh-operator
            control-plane: controller-manager
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - operator
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=debug
                - --region=local
                - --environment=local
                - --image-tag=0.0.0
                - --image-name="ghcr.io/platform-mesh/platform-mesh-operator"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
                - --tracing-enabled=true
                - --tracing-config-service-name=platform-mesh-operator.NAMESPACE
                - --tracing-config-service-version=0.0.0
                - --tracing-config-collector-endpoint=test:4317
              env: null
              image: ghcr.io/platform-mesh/platform-mesh-operator:0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: platform-mesh-operator
              ports:
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: platform-mesh-operator
          terminationGracePeriodSeconds: 10
operator match the snapshot:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: platform-mesh-operator
    rules:
      - apiGroups:
          - core.platform-mesh.io
        resources:
          - platformmeshes
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - core.platform-mesh.io
        resources:
          - platformmeshes/finalizers
        verbs:
          - update
      - apiGroups:
          - core.platform-mesh.io
        resources:
          - platformmeshes/status
        verbs:
          - get
          - patch
          - update
      - apiGroups:
          - ""
        resources:
          - secrets
        verbs:
          - get
          - list
          - watch
          - create
          - delete
          - update
          - patch
      - apiGroups:
          - delivery.ocm.software
        resources:
          - componentversions
          - resources
          - fluxdeployers
        verbs:
          - get
          - list
          - watch
          - create
          - delete
          - update
          - patch
      - apiGroups:
          - helm.toolkit.fluxcd.io
        resources:
          - helmreleases
        verbs:
          - get
          - list
          - watch
          - create
          - delete
          - update
          - patch
      - apiGroups:
          - operator.kcp.io
        resources:
          - rootshards
          - frontproxies
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - cert-manager.io
        resources:
          - certificates
        verbs:
          - get
          - list
          - watch
          - create
          - delete
          - update
          - patch
      - apiGroups:
          - source.toolkit.fluxcd.io
        resources:
          - ocirepositories
          - helmreleases
        verbs:
          - get
          - list
          - create
          - update
          - patch
          - delete
          - watch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: platform-mesh-operator
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: platform-mesh-operator
    subjects:
      - kind: ServiceAccount
        name: platform-mesh-operator
        namespace: NAMESPACE
  3: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: platform-mesh-operator
      namespace: NAMESPACE
    spec:
      replicas: 0
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: platform-mesh-operator
      strategy:
        rollingUpdate:
          maxSurge: 5
          maxUnavailable: 0
        type: RollingUpdate
      template:
        metadata:
          annotations:
            sidecar.istio.io/inject: "true"
          labels:
            app: platform-mesh-operator
            control-plane: controller-manager
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - operator
                - --leader-elect
                - --metrics-bind-address=:9090
                - --health-probe-bind-address=:8090
                - --log-level=debug
                - --region=local
                - --environment=local
                - --image-tag=0.0.0
                - --image-name="ghcr.io/platform-mesh/platform-mesh-operator"
                - --shutdown-timeout=1m
                - --max-concurrent-reconciles=10
              env: null
              image: ghcr.io/platform-mesh/platform-mesh-operator:0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 1
                httpGet:
                  path: /healthz
                  port: 8090
                periodSeconds: 10
              name: platform-mesh-operator
              ports:
                - containerPort: 9090
                  name: metrics
                  protocol: TCP
                - containerPort: 8090
                  name: health-port
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8090
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  memory: 512Mi
                requests:
                  cpu: 40m
                  memory: 50Mi
              securityContext:
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                seccompProfile:
                  type: RuntimeDefault
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /readyz
                  port: 8090
                periodSeconds: 10
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: platform-mesh-operator
          terminationGracePeriodSeconds: 10
  4: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: Role
    metadata:
      name: platform-mesh-operator
    rules:
      - apiGroups:
          - coordination.k8s.io
        resources:
          - leases
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
      - apiGroups:
          - ""
        resources:
          - pods
        verbs:
          - list
          - delete
      - apiGroups:
          - ""
        resources:
          - events
        verbs:
          - create
          - delete
          - get
          - list
          - patch
          - update
          - watch
  5: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: platform-mesh-operator
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: Role
      name: platform-mesh-operator
    subjects:
      - kind: ServiceAccount
        name: platform-mesh-operator
        namespace: NAMESPACE
  6: |
    apiVersion: v1
    imagePullSecrets:
      - name: github
    kind: ServiceAccount
    metadata:
      name: platform-mesh-operator
      namespace: NAMESPACE
